docker/test_env/ethereum_env.go

```diff
@@ -441,38 +542,50 @@ func (e *EthereumNetworkContainers) wasAlreadyRestarted(containerName string) bo
 	return false
 }
 
-var restartedContainers = make(EthereumNetworkContainers, 0)
+func createHostDirectories() (string, string, error) {
+	customConfigDataDir, err := os.MkdirTemp("", "custom_config_data")
+	if err != nil {
+		return "", "", err
+	}
 
-func (b *EthereumNetwork) setExistingContainerName(ct ContainerType) EnvComponentOption {
-	if !b.isRecreated {
-		return func(c *EnvComponent) {}
+	valKeysDir, err := os.MkdirTemp("", "val_keys")
+	if err != nil {
+		return "", "", err
 	}
 
-	// in that way we can support restarting of multiple nodes out of the box
-	for _, container := range b.Containers {
-		if container.ContainerType == ct && !restartedContainers.wasAlreadyRestarted(container.ContainerName) {
-			restartedContainers.add(container)
-			return func(c *EnvComponent) {
-				if container.ContainerName != "" {
-					c.ContainerName = container.ContainerName
-				}
+	return customConfigDataDir, valKeysDir, nil
+}
+
+func waitForChainToFinaliseFirstEpoch(lggr zerolog.Logger, evmClient blockchain.EVMClient) error {
```

Comment:
How long does this usually take?

Reply:
depends on the chain settings (seconds per slot + slots per epoch), with some reasonable ones (8 and 4 respectively) it might take around 1 - 1.5 minutes,

Reply:
But maybe we shouldn't make that wait the default? And instead explicitly requested?